internal/services/housekeeper/tablePartitionCleaner.go

```diff
@@ -0,0 +1,33 @@
+package housekeeper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/redhatinsights/ros-ocp-backend/internal/config"
+	database "github.com/redhatinsights/ros-ocp-backend/internal/db"
+)
+
+func DeletePartitions() {
+	cfg := config.GetConfig()
+	db := database.GetDB()
+	currentTime := time.Now()
+
+	// subtracting $cfg.DataRetentionPeriod from the currentTime
+	retentionThresholdDate := currentTime.AddDate(0, 0, -cfg.DataRetentionPeriod)
+
+	// IF $retentionThresholdDate date(day) is less than 15th of the month then
+	// $partitionTableDate i.e. partition which are marked to be delete should be before 16th of the pervious month.
```

Comment:
`$partitionTableDate..... should be before 16th of the pervious month.` is a bit confusing. We are using this variable to delete the older partitions than the value and not that we are storing an older date than the 16th.

We can modify it like,
`If the day of the month in $retentionThresholdDate is less than 15,
set $partitionTableDate to the 16th of the previous month.
Otherwise, set $partitionTableDate to the 1st of the current month.`